File: API/app/Models/Members.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Laravel\Sanctum\HasApiTokens;


class Members extends Model
{
    use HasFactory, HasApiTokens;
    protected $fillable= [
        'id_users', 'nama_tempat', 'lokasi', 'no_wa'
    ];

    public function users_member(): BelongsTo
    {
        return $this->belongsTo(User::class, 'id_users');
    }

    public function isMember()
    {
        return true;
    }
}

File: lib/models/members.dart
import 'package:frontend/models/users.dart';

class Member {
  final int id;
  final int id_users;
  final String nama_tempat;
  final String lokasi;
  final String no_wa;
  final Users user;

  Member(
      {required this.id,
      required this.id_users,
      required this.nama_tempat,
      required this.lokasi,
      required this.no_wa,
      required this.user});

  factory Member.fromJson(Map<String, dynamic> json) {
    return Member(
        id: json['id'] ?? 0,
        id_users: json['id_users'] ?? 0,
        nama_tempat: json['nama_tempat'] ?? '',
        lokasi: json['lokasi'] ?? '',
        no_wa: json['no_wa'] ?? '',
        user: Users.fromJson(json['user'] ?? {}));
  }
}
